HNParsePage.py

```diff
@@ -14,7 +14,10 @@ def __init__(self, data, old_style=False):
         self.offset = 0
         def Text(self, code):
             try:
-                self.characters.append(bytes([self.data[self.offset+5],self.data[self.offset+4]]).decode("gbk"))
+                if len(self.data) > self.offset+5:
+                    self.characters.append(bytes([self.data[self.offset+5],self.data[self.offset+4]]).decode("gbk"))
```

Comment:
Yes,  a lot of this code is work in progress. Probably should be totally rewritten at some point.